py/desisim/io.py

```diff
@@ -500,6 +500,98 @@ def read_templates(wave, objtype, nspec=None, randseed=1, infile=None):
     return outflux, outmeta
     
 
+def read_base_templates(objtype='ELG', observed=False, emlines=False):
+    """Return the base, rest-frame, spectral continuum templates for each objtype.
+
+    The appropriate environment variable must be set depending on OBJTYPE.  For example,
+    DESI_ELG_TEMPLATES, DESI_LRG_TEMPLATES, etc., otherwise an exception will be raised.
+
+    Args:
+      objtype (str, optional): object type to read (ELG, LRG, QSO, BGS, STD, or STAR;
+        defaults to 'ELG').
+      observed (bool): Read the observed-frame templates (defaults to False).
+      emlines (bool): Read the spectral templates which include emission lines (defaults
+        to False; only applies to object types ELG and BGS).
+
+    Returns:
+      flux (numpy.ndarray): Array [ntemplate,npix] of flux values [erg/s/cm2/A].
+      wave (numpy.ndarray): Array [npix] of wavelengths for FLUX [Angstrom].
+      meta (astropy.Table): Meta-data table for each object.  The contents of this
+        table varies depending on what OBJTYPE has been read.
+
+    Raises:
+      EnvironmentError: If the appropriate environment variable is not set.
+      IOError: If the base templates are not found.
+    
+    """
+    from astropy.io import fits
+    from astropy.table import Table
+    from desispec.io.util import header2wave
+
+    otype = objtype.upper()
+    if otype=='FSTD':
+        otype = 'STAR'
+
+    key = 'DESI_'+otype+'_TEMPLATES'
+    if key not in os.environ:
+        log.error('Required ${} environment variable not set'.format(key))
+        raise EnvironmentError
+
+    objfile = os.getenv(key)
+
+    # Handle special cases for the ELG & BGS templates.
+    if otype=='ELG' or otype=='BGS':
+        if observed:
+            objfile = objfile.replace('templates_','templates_obs_')
+        elif emlines is not True:
+            objfile = objfile.replace('templates_','continuum_templates_')
+
+    if os.path.isfile(objfile):
+        log.info('Reading {}'.format(objfile))
+    else: 
+        log.error('Base templates file {} not found'.format(objfile))
+        raise IOError()
+
+    flux, hdr = fits.getdata(objfile, 0, header=True)
+    meta = Table(fits.getdata(objfile, 1))
+    wave = header2wave(hdr)
+
+    return flux, wave, meta
+
+def write_templates(flux, wave, meta, objtype, outfile=None, comments=None,
+                    units=None, header_comments=None):
+    """Write out simulated galaxy templates.  (Incomplete documentation...)
```

Comment:
In `desispec.io`, the standard is `write_blat(outfile, data...)` rather than `write_blat(data, ..., outfile)`.  The use of `outfile=None` here appears to be preparing for the possibility that someone would call this without giving the output filename, but that just causes the current code to crash.  Is there a planned case for when not specifying `outfile` would be useful?  If not, please reorder to make outfile the first required parameter for consistency with other desihub code.